### PR TITLE
feat: Break up webhooks from onEvent

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Async handlers lagging happens occasionally and the problem could be either with webhooks or with exports, separating them out would make debugging and handling easier at the cost of some extra processing and pods. Furthermore webhooks might be much faster or slower to handle compared to exports so the waiting together might not be best

see also https://posthog.slack.com/archives/C0374DA782U/p1688474721871699?thread_ts=1688246723.752029&cid=C0374DA782U

The plan is:
1. ship this (it's a no-op for prod)
2. ship a follow-up PR to add capabilities (this one is too big already) - still a no-op for prod
3. kill current async deployment
4. create 2 new consumers, which we'll call `-apps-on-event` and `-webhooks` respectively & set the kafka offsets for both to match existing async kafka offsets
5. start the 2 separate deployments for onEvent and webhooks respectively

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts [hunk 1]
```diff
@@ -4,15 +4,21 @@ import { KAFKA_EVENTS_JSON, prefix as KAFKA_PREFIX } from '../../config/kafka-to
...
-import { eachBatchAsyncHandlers } from './batch-processing/each-batch-async-handlers'
+import {
+    eachBatchAppsOnEventHandlers,
+    eachBatchAsyncHandlers,
+    eachBatchWebhooksHandlers,
+} from './batch-processing/each-batch-async-handlers'
 import { KafkaJSIngestionConsumer } from './kafka-queue'
 
-export const startOnEventHandlerConsumer = async ({
+export const startAsyncHandlerConsumer = async ({
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/on-event-handler-consumer.ts:14]: Personal preference: I would keep the start consumer functions separate for each of the `kind`. That way we have the option of making e.g. the webhooks interface much narrower i.e. maybe we could get away with only passing the postgres client in, and thereby avoid needing to load plugins, or clickhouse client for instance.

I would tend to prefer that pattern over pushing the detail into if statements within the function.

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts [hunk 2]
```diff
@@ -7,23 +7,83 @@ import { runInstrumentedFunction } from '../../utils'
 import { KafkaJSIngestionConsumer } from '../kafka-queue'
 import { eachBatch } from './each-batch'
 
+// TODO: remove once we've migrated
 export async function eachMessageAsyncHandlers(message: KafkaMessage, queue: KafkaJSIngestionConsumer): Promise<void> {
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts:11]: personal preference: I'm not a fan of the IngestionConsumer abstraction and would look at how to remove it at some point.

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts [hunk 3]
```diff
@@ -7,23 +7,83 @@ import { runInstrumentedFunction } from '../../utils'
...
     const clickHouseEvent = JSON.parse(message.value!.toString()) as RawClickHouseEvent
     const event = convertToIngestionEvent(clickHouseEvent)
 
+    await Promise.all([
+        runInstrumentedFunction({
+            server: queue.pluginsServer,
+            event: event,
+            func: () => queue.workerMethods.runAppsOnEventPipeline(event),
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts:19]: tangential: This workerMethods dependency is a strange and confusing feature of the events ingestion imo. No action required though.

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts [hunk 4]
```diff
@@ -7,23 +7,83 @@ import { runInstrumentedFunction } from '../../utils'
...
+        }),
+    ])
+}
+
+// TODO: remove once we've migrated
+export async function eachBatchAsyncHandlers(
+    payload: EachBatchPayload,
+    queue: KafkaJSIngestionConsumer
+): Promise<void> {
+    await eachBatch(payload, queue, eachMessageAsyncHandlers, groupIntoBatches, 'async_handlers')
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts:40]: tangential: the signature of these eachBatch functions is very convoluted imo. Having eachBatchAsyncHandlers just perform the batching, and running the processing directly would be nicer imo

no action required.

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts (hunk 3 above)
Comment (tiina303) [reply plugin-server/src/main/ingestion-queues/batch-processing/each-batch-async-handlers.ts:19]: would be nice to do a clean-up, but maybe we just wait until we switch to the batching world for exports and persons

**Review by hazzadous — APPROVED** [review]
Looks good. More simplifications to be done but it's a start!

**Pull request merged**